documentation/specs/src/interoperability/ethereum-bridge/proofs.md

```diff
@@ -1,101 +1,83 @@
+      validator set of some epoch $E' = E + 1$ must be signed by a quorum of
+      Namada validators from $E$ before the end of this epoch.
+    - The process of updating the set of validators in Ethereum can be thought
+      of as a "transfer of power", from the old set of validators to the new one.
+      Note that the validators in the set may be the same between epochs, but this
+      must be communicated to the Ethereum bridge smart contracts, either way.
+    - Validator sets in the `Bridge` contract rotate in a pipeline. This contract
+      always keeps track of the current set of validators, as well as the next set
+      of validators. Initially, the current and next set of validators are initialized
+      with the same value. Subsequent updates to the contract will shift the next set
+      of validators to the current set, and change the value of the next set to the
+      set of validators that was freshly signed by a quorum of Namada validators.
+
+More information on how these proofs are constructed is available
+under the [Ethereum events attestation section].
+
+[Ethereum events attestation section]: ethereum_events_attestation.md#vote-extension-protocol-transactions
+
+### Recovering from a validator set update failure
 
 If vote extensions are not available, we cannot guarantee that a quorum of 
 validator signatures can be gathered for the message that updates the 
-validator set before the epoch ends.
+validator set before some epoch $E$ ends. Should this scenario take place,
+the Ethereum bridge will halt in the Namada to Ethereum direction, since
+it will not be able to authenticate user transfers.
 
-If a significant number of validators become inactive in the next epoch, we 
-need a means to complete validator set update. Until this is done, the 
-bridge will halt. 
+In this case, the validators from $E$ will need to coordinate, offline, the
+crafting of a transaction including their vote on the validator set of $E + 1$.
+The only way this is impossible is if more than $1/3$ (by stake) of the consensus
+validators from $E$ delete their Ethereum keys, which is extremely unlikely.
 
-In this case, the validators from that epoch will need to craft a 
-transaction with a quorum of signatures offline and submit it on-chain. This 
-transaction should include the validator set update. 
+## Namada Bridge Relayers
 
-The only way this is impossible is if more than 1/3 of the validators by 
-stake from that epoch delete their ethereum keys, which is extremely unlikely.
+Proofs themselves do not alter any state in Ethereum. Rather, they
+authenticate state transitions in Ethereum based on the security
+properties (in the BFT sense) of the Namada network. Carrying out
+state updates in Ethereum is the task of an __Ethereum bridge relayer__.
+
+The job of an Ethereum bridge relayer is to submit proofs to the Ethereum
+bridge smart contracts before the conclusion of each Namada epoch. The default
```

Comment:
Do proofs specifically have to be submitted before the end of the epoch? Do you mean submitting batches before the validator set changes?

Reply:
yeah, exactly, so you can still authenticate stuff

Reply:
aye - we should just clarify that in this sentence then